packages/advisor-components/src/RuleDetails/RuleDetails.tsx

```diff
@@ -0,0 +1,247 @@
+
+import React, { MouseEventHandler, useContext } from 'react';
+import { Link } from 'react-router-dom';
+import { useIntl } from 'react-intl';
+import Markdown from 'react-markdown';
+import rehypeRaw from 'rehype-raw';
+import rehypeSanitize from 'rehype-sanitize';
+
+import { Split, SplitItem, Stack, StackItem, Text, TextContent, TextVariants } from '@patternfly/react-core';
+import ExternalLinkAltIcon from '@patternfly/react-icons/dist/js/icons/external-link-alt-icon';
+
+import InsightsLabel from '@redhat-cloud-services/frontend-components/InsightsLabel';
+import { SeverityLine } from '@redhat-cloud-services/frontend-components-charts/SeverityLine';
+
+import { IMPACT_LABEL_KEY, LIKELIHOOD_LABEL_KEY, RISK_OF_CHANGE_LABEL_KEY, TOTAL_RISK_LABEL_KEY } from '../constants';
+import RebootRequired from '../RebootRequired/RebootRequired';
+import RuleRating from '../RuleRating/RuleRating';
+import ContextWrapper, { ContextWrapperProps, ProductContext } from '../ContextWrapper';
+import messages from '../messages';
+import { barDividedList, topicLinks } from '../common';
+import { RiskOfChange } from '../types';
+import { Likelihood } from '../types';
+import { TopicRhel } from '../types';
+import { Rating } from '../types';
+import { RuleContentOcp } from '../types';
+import { RuleContentRhel } from '../types';
+import { AdvisorProduct } from '../types';
+import { string } from 'prop-types';
+
+interface RuleDetailsBaseProps {
+  rule: RuleContentOcp | RuleContentRhel;
+  resolutionRisk?: number;
+  resolutionRiskDesc?: string;
+  header?: React.ReactNode;
+  isDetailsPage: boolean;
+  topics: TopicRhel[];
+  /**
+   * onVoteClick - a callback used to update the rating of a particular rule
+   * @param {string} ruleId - ID (usually in plugin|error_key format) of the rule that needs to be updated
+   * @param {number} newRating rating (-1, 0, 1)
+   */
+  onVoteClick?: (ruleId: string, calculatedRating: Rating) => unknown;
+  onViewAffectedClick?: (ruleId: string) => unknown;
+}
+
+interface RuleDetailsProps extends RuleDetailsBaseProps, ContextWrapperProps {}
+
+const RuleDetailsBase: React.FC<RuleDetailsBaseProps> = (props) => {
+  const intl = useIntl();
+  const product = useContext(ProductContext);
+  const { header, rule, isDetailsPage, topics, onVoteClick, onViewAffectedClick, resolutionRisk, resolutionRiskDesc, children } = props;
+
+  const renderDescription = () => {
+    const ruleDescription = (data: string, isGeneric = false) =>
+      typeof data === 'string' &&
+      Boolean(data) && (
+        <span className={isGeneric ? 'genericOverride' : ''}>
+          <Markdown rehypePlugins={[rehypeRaw, rehypeSanitize]}>{data}</Markdown>
+        </span>
+      );
+
+    switch (product) {
+      case AdvisorProduct.ocp:
+        return ruleDescription(rule.generic, true);
+        break;
+      case AdvisorProduct.rhel:
+        return isDetailsPage ? ruleDescription(rule.generic, true) : ruleDescription(rule.summary);
+    }
+  };
```

Comment:
This is a simple `if` case. I am not a big fan of `switch` statements, can we use that instead?